### PR TITLE
Update API doc and user manual to reflect group feature

### PR DESCRIPTION
**Pull request by philip-cc**
### Description
This PR updates the API documentation and user manual to include the new group feature. 

### Relevant Issues/PRs
Issue #241 
PR #237 

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [ ] Refactor
- [ ] Bugfix
- [x] Updated documentation
- [ ] Kubernetes changes
- [ ] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [ ] I have pulled in the latest changes from master.
- [ ] I have run `make lint`.
- [ ] I have successfully run `make tests`.
- [ ] I have updated relevant CI/CD workflows.
- [ ] I have updated relevant Kubernetes manifests/scripts.
- [x] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [ ] I have added the license to new source code files.
- [ ] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: documentation/api/api-v3.1.0.md [hunk 1]
```diff
@@ -290,46 +261,132 @@ the newly created user.
...
+defining which endpoints the group has access to. Possible scopes are `READ`, `CREATE`, `INDEX`,
+`OBJECTPERMISSIONS`, and `USERMANAGEMENT`. Requires the scope `USERMANAGEMENT`.
+
+| Name     | Type         | Description                                       |
+|----------|--------------|---------------------------------------------------|
+| `scopes` | []enum Scope | An array of scopes the newly created group posses |
+
+### `authn.CreateGroupResponse`
+The structure returned by a `authn.CreateGroup` request. It contains the Group ID of the newly
+created user.
```
[the remark this answers is not in the document]
Comment (philip-cc) [reply documentation/api/api-v3.1.0.md]: Fixed

File: documentation/api/api-v3.1.0.md [hunk 2]
```diff
@@ -290,46 +261,132 @@ the newly created user.
...
+groups with access to the Object specified in the request.
+
+| Name        | Type     | Description           |
+|-------------|----------|-----------------------|
+| `group_ids` | []string | An array of group IDs |
+
+### `authz.AddPermissionRequest`
+The structure used as an argument for an `authz.AddPermission` request. It contains the ID of the
+Object the client wishes to add permissions to and the Group ID of the group to be added to the access
+list. Requires the scope `OBJECTPERMISSIONS`.
```
[the remark this answers is not in the document]
Comment (philip-cc) [reply documentation/api/api-v3.1.0.md]: Fixed

File: documentation/api/api-v3.1.0.md [hunk 3]
```diff
@@ -290,46 +261,132 @@ the newly created user.
...
+| `object_id` | string | The object                        |
+| `target`    | string | The target for permission change  |
+
+### `authz.AddPermissionResponse`
+The structure returned by a `authz.AddPermission` request. The structure is empty.
+
+### `authz.RemovePermissionRequest`
+The structure used as an argument for a `authz.RemovePermission` request. It contains the ID of
+the Object the client wishes to remove permissions from and the Group ID of the group to be removed
+from the access list. Requires the scope `OBJECTPERMISSIONS`.
```
[the remark this answers is not in the document]
Comment (philip-cc) [reply documentation/api/api-v3.1.0.md]: Fixed

File: documentation/api/api-v3.1.0.md [hunk 4]
```diff
@@ -390,61 +449,90 @@ returns the decrypted content.
...
 ```
 
-## `authz.AddPermission`
+### `authn.LoginUser`
 
-Adds a User to the access list of the specified object. This call can fail if the caller
-does not have access to the object, if the target user does not exist, or if the Storage
-Service cannot reach the auth storage. In these cases, an error is returned.
+Logs in an existing user, returning a User Access Token. Note that this token is valid for 1 hour.
+This call can fail if the caller the wrong credentials or if the Auth Service cannot reach the auth
```
[the remark this answers is not in the document]
Comment (philip-cc) [reply documentation/api/api-v3.1.0.md]: Fixed

File: documentation/api/api-v3.1.0.md [hunk 5]
```diff
@@ -390,61 +449,90 @@ returns the decrypted content.
...
 
-## `authn.RemoveUser`
+## `authz`
 
-Deletes an existing user. This call can fail if the caller is lacking the required scope, if the
-user does not exist, or if the Auth Service cannot reach the auth storage, in which case an error is
-returned.
+### `authz.GetPermissions`
+
+Returns a list of users with access to the specified object. This call can fail if the
```
[the remark this answers is not in the document]
Comment (philip-cc) [reply documentation/api/api-v3.1.0.md]: Fixed

File: documentation/api/api-v3.1.0.md [hunk 6]
```diff
@@ -390,61 +449,90 @@ returns the decrypted content.
...
+be authenticated and authorized in order to get the object permissions.
 
 ```
-rpc RemoveUser (RemoveUserRequest) returns (RemoveUserResponse)
+rpc GetPermissions (GetPermissionsRequest) returns (GetPermissionsResponse)
+```
+
+### `authz.AddPermission`
+
+Adds a User to the access list of the specified object. This call can fail if the caller
```
[the remark this answers is not in the document]
Comment (philip-cc) [reply documentation/api/api-v3.1.0.md]: Fixed

File: documentation/api/api-v3.1.0.md [hunk 7]
```diff
@@ -390,61 +449,90 @@ returns the decrypted content.
...
+does not have access to the object, if the target user does not exist, or if the Storage
+Service cannot reach the auth storage. In these cases, an error is returned.
+
+```
+rpc AddPermission (AddPermissionRequest) returns (ReturnCode)
+```
+
+### `authz.RemovePermission`
+
+Removes a User from the access list of the specified object. This call can fail if the
```
[the remark this answers is not in the document]
Comment (philip-cc) [reply documentation/api/api-v3.1.0.md]: Fixed

File: documentation/api/api-v3.1.0.md [hunk 8]
```diff
@@ -290,46 +261,132 @@ the newly created user.
...
+The structure used as an argument for a `authz.GetPermissions` request. It contains the ID of the
+Object the client wishes to get the permission list for. Requires the scope `OBJECTPERMISSIONS`.
+
+| Name        | Type   | Description           |
+|-------------|--------|-----------------------|
+| `object_id` | string | The object identifier |
+
+### `authz.GetPermissionsResponse`
+The structure returned by a `storage.GetPermissions` request. It contains a list of group IDs of
+groups with access to the Object specified in the request.
```
[the remark this answers is not in the document]
Comment (philip-cc) [reply documentation/api/api-v3.1.0.md:343]: As discussed, this is explained in the manual

**Review by nela-cc — APPROVED** [review]

**Pull request merged**